packages/babel-preset/index.js

```diff
@@ -73,15 +73,10 @@ module.exports = function shopifyCommonPreset(
     // Enable loose mode to use assignment instead of defineProperty when typescript is enabled
     // class-properties are handled by preset-env
     // But when using typescript we need to transpile them in loose mode to support proposal-decorators's legacy mode
-    typescript && [
-      require.resolve('@babel/plugin-proposal-class-properties'),
-      {loose: true},
-    ],
+    typescript && require.resolve('@babel/plugin-proposal-class-properties'),
```

Comment:
https://babeljs.io/blog/2021/04/29/7.14.0#new-class-features-enabled-by-default suggests plugin-proposal-class-properties and plugin-proposal-private-methods are now both enabled by preset-env. Thus we might be able stop referencing manually including these plugins entirely

However https://babeljs.io/docs/en/babel-plugin-proposal-decorators#note-compatibility-with-babelplugin-proposal-class-properties suggest that if you are including your plugins manually and using `@babel/plugin-proposal-class-properties`, then `@babel/plugin-proposal-decorators` must come before `@babel/plugin-proposal-class-properties`.

How do those two statements interact? Are we able to remove these two explicit plugin references (deleting this and the next line, or does the usage of `@babel/plugin-proposal-decorators` force us to always include these lines)

---


I wonder if we should add a new option `decorators` and make support for them (and the enabling of the assumptions) opt-in. They're non-standard, stuck at stage-2 since forever, experimental, and force transpilation of other standard features in a non-standard way. That doesn't feel like an approach that we should be advocating for general adoption by enabling it by default.